### PR TITLE
FE: Disable invalid values source options if parameter has linked filters

### PR DESCRIPTION
**Pull request by calherries**
Epic: https://github.com/metabase/metabase/issues/34604

Part 3 out of 3 for #33892, see comment [here](https://github.com/metabase/metabase/issues/33892#issuecomment-1752488489) for the plan.

Part 1 is here: #34424
Part 2 is here: #34438

This PR prevents users from setting a filter's values source to anything other than "From connected fields" if the filter has any linked filters. It does so by disabling the "Edit" button here and showing an explanation in the tooltip:

image

### CODE REVIEW

File: frontend/src/metabase/parameters/components/ValuesSourceSettings/ValuesSourceSettings.tsx [hunk 1]
```diff
@@ -62,38 +64,68 @@ const ValuesSourceSettings = ({
   );
 };
 
+function isOnlyConnectedFieldSourceAllowed(parameter: Parameter) {
```
Comment (ranquild) [inline frontend/src/metabase/parameters/components/ValuesSourceSettings/ValuesSourceSettings.tsx:67]: The name and the function body is a bit hard to follow. I'd change it to something like:
```
function hasLinkedFilters({ filteringParameters }: Parameter) {
  return filteringParameters != null && filteringParameters.length > 0;
}
```

and later use it like this:
```
isEditDisabled: hasLinkedFilters(parameter)
```

File: frontend/src/metabase/parameters/components/ValuesSourceSettings/ValuesSourceSettings.tsx [hunk 2]
```diff
@@ -62,38 +64,68 @@ const ValuesSourceSettings = ({
...
   isSelected?: boolean;
   onEditClick?: () => void;
+  isEditDisabled?: boolean;
 }
 
 const RadioLabel = ({
   title,
   isSelected,
   onEditClick,
+  isEditDisabled,
```
Comment (ranquild) [inline frontend/src/metabase/parameters/components/ValuesSourceSettings/ValuesSourceSettings.tsx:83]: set default value here `isEditDisabled = false` to avoid using `!!` later

File: frontend/src/metabase/parameters/components/ValuesSourceSettings/ValuesSourceSettings.unit.spec.tsx [hunk 3]
```diff
@@ -48,4 +48,46 @@ describe("ValuesSourceSettings", () => {
       values: ["A"],
     });
   });
+
+  it("Editing the values source should be disabled when the filter has linked filters", () => {
```
Comment (ranquild) [inline frontend/src/metabase/parameters/components/ValuesSourceSettings/ValuesSourceSettings.unit.spec.tsx:52]: we usually don't sentence-case test names, it should be `editing the values ...`

**Review by ranquild — APPROVED** [review]
Please take a look at comments before merging

**Comment by calherries** [comment]
@ranquild implemented your suggestions, thanks
